src/it/resources/com/google/checkstyle/test/chapter4formatting/rule487modifiers/InputModifierOrder.java

```diff
@@ -79,24 +83,25 @@ strictfp private void doStuff()
 
   /** holder for redundant 'public' modifier check. */
   public static interface InputRedundantPublicModifier {
-    public void a();
+    public void abc1();
 
-    void b();
+    void abc2();
 
-    abstract void c();
+    abstract void abc3();
 
     public float PI_PUBLIC = (float) 3.14;
 
     final float PI_FINAL = (float) 3.14;
 
-    /** all OK */
+    /** all OK. */
     float PI_OK = (float) 3.14;
   }
 
   // violation below ''private' modifier out of order with the JLS suggestions.'
   final private void method() {}
 }
 
+// violation below 'Top-level class RedundantFinalClass has to reside in its own source file.'
 final class RedundantFinalClass {
```

Comment:
This input file had many extra classes and interfaces. I tried moving them inside the main class but some class had static variables so the code became not compile-able. So I abandoned the idea of moving these classes into the main class

Reply:
I am ok to keep this violation in Inputs , it will be frequent, we need this to keep required types embedded in single file.